readthedocs/core/unresolver.py

```diff
@@ -53,6 +59,24 @@ class UnresolvedURL:
     external: bool = False
 
 
+class OriginType(Enum):
+
+    """From where the custom domain was resolved."""
+
+    custom_domain = auto()
+    public_domain = auto()
+    external_domain = auto()
+    http_header = auto()
+
+
+@dataclass(slots=True)
+class UnresolvedDomain:
+    origin: OriginType
+    project: Project
+    domain: Domain = None
+    external_version_slug: str = None
```

Comment:
It seems weird that we're passing this around. Do we not just want... an optional `version` here? Or perhaps `force_version`? Not 100% on this though. 

It's basically just for single version projects so we can resolve the PR to the right version, right? Non-single version PR builds use the path in the URL as well?

Reply:
Agree with Eric here. It sounded weird to me as well: "How the _external version slug_ is related to a Domain?". I'm not seeing that clearly in this context.

Reply:
Maybe the name of the object is not accurate and should be something like `UnresolvedPath` or `UnresolvedURL` instead of "Domain"?

Reply:
Ahhh, gotcha! We are getting the external version slug from the domain itself! I forget we were including it there 👍🏼 . It makes sense now to me.

Reply:
> It's basically just for single version projects so we can resolve the PR to the right version, right? Non-single version PR builds use the path in the URL as well?

Yes, this is to resolve the version of single version PRs. We are currently forcing the version from the domain to match the version from the path as well.